### PR TITLE
`aws_ebs_snapshot_import`: init resource

### PR DESCRIPTION
**Pull request by grahamc**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
* **New Resource:** aws_ebs_snapshot_import  (#16373)
```

Output from acceptance testing:

```
$ time make fmt testacc TESTARGS='-run=TestAccAWSEBSSnapshotImport'
==> Fixing source code with gofmt...
gofmt -s -w ./aws ./awsproviderlint/helper ./awsproviderlint/main.go ./awsproviderlint/passes
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSEBSSnapshotImport -timeout 120m
=== RUN   TestAccAWSEBSSnapshotImport_basic
=== PAUSE TestAccAWSEBSSnapshotImport_basic
=== CONT  TestAccAWSEBSSnapshotImport_basic
--- PASS: TestAccAWSEBSSnapshotImport_basic (625.66s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	25.694s

real	12m33.506s
user	2m43.535s
sys	2m7.373s

```

---

I don't feel incredibly comfortable that the code I wrote is good or totally right, but I gave it my best. I'm not much of a Go programmer, and started with the ebs_snapshot resource. Note I didn't add support for importing, because it seems the data around the import parameters are "Disappeared" once the import task completes.

My use case here is combining aws_ebs_snapshot_import with aws_ami and aws_ami_copy to upload, register, and distribute AMIs to a variety of regions. I use Terraform to support the pruning of images over time.

### CODE REVIEW

File: aws/resource_aws_ebs_snapshot_import_test.go
Comment (grahamc) [inline aws/resource_aws_ebs_snapshot_import_test.go:188]: This is a bit funny probably, but to test importing a disk image we need a disk image to import. The image on disk, base64'd, is about 3400 bytes long. Compressing, then base64'ing is much shorter, with the trade-off that we need to decompress and rebase64 on the fly for the test.

File: aws/resource_aws_ebs_snapshot_import.go
Comment (grahamc) [inline aws/resource_aws_ebs_snapshot_import.go:281]: This doesn't seem to be following the spirit of the rule, but I'm not confident in the proper way to handle this case.

**Comment by dhess** [comment]
This would be wonderful!

File: aws/resource_aws_ebs_snapshot_import.go
Comment (grahamc) [inline aws/resource_aws_ebs_snapshot_import.go:161]: It seems this cannot create a resource with tags, but it can add tags later. Todo: look in to this.

```
Error: error importing EBS Snapshot: InvalidParameter: 'snapshot' is not a valid taggable resource type for this operation.
	status code: 400, request id: d49f7fd0-4c06-48bb-a976-8ea12b9ed8ef
```

**Comment by grahamc** [comment]
Just rebased this on top of master, and with it I added tags to the test case after discovering the previous revision of this PR didn't correctly tag the snapshot.

**Comment by grahamc** [comment]
Rebased and re-ran the tests.

**Comment by m4h3** [comment]
can this get some attention ?

**Comment by grahamc** [comment]
@m4h3 make sure to :+1: the PR itself and it'll get slightly more bumped in their priority order. I build my own AWS terraform provider using Nix to get around the backlog here.

**Comment by grahamc** [comment]
I heard from a Hashicorp person that this PR might get looked at soon.

File: aws/resource_aws_ebs_snapshot_import_test.go
Comment (DrFaust92) [inline aws/resource_aws_ebs_snapshot_import_test.go:21]: lets use the more common `rName := acctest.RandomWithPrefix("tf-acc-test")`

File: aws/resource_aws_ebs_snapshot_import.go
Comment (DrFaust92) [inline aws/resource_aws_ebs_snapshot_import.go:208]: same as above

File: aws/resource_aws_ebs_snapshot_import.go
Comment (DrFaust92) [inline aws/resource_aws_ebs_snapshot_import.go:90]: `validation.StringInSlice(ec2.DiskImageFormat_Values(), false ),`

File: aws/resource_aws_ebs_snapshot_import.go
Comment (DrFaust92) [inline aws/resource_aws_ebs_snapshot_import.go:147]: according to [Docs](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_ImportSnapshot.html) default value is `vmimport`

File: aws/resource_aws_ebs_snapshot_import.go
Comment (DrFaust92) [inline aws/resource_aws_ebs_snapshot_import.go:24]: i wonder if these are needed and can use existing snapshot function similar to AMI resources. see `resource_aws_ami_copy` for instance

**Review by DrFaust92 — CHANGES_REQUESTED** [review]
Not a maintainer but want to help move this along

also add [disappears-acceptance-tests](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/running-and-writing-acceptance-tests.md#disappears-acceptance-tests)

**Comment by DrFaust92** [comment]
also add [changelog file](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/pullrequest-submission-and-lifecycle.md#changelog-process)

File: aws/resource_aws_ebs_snapshot_import.go
Comment (cole-h) [reply aws/resource_aws_ebs_snapshot_import.go:24]: I took a look a couple days ago and didn't think that it would be worth it to pursue further. There were enough differences that (IIRC) it wouldn't be drop-in.

**Comment by grahamc** [comment]
Thanks, @DrFaust92! I've had @cole-h help me out and address your feedback. I also rebased on `main` and it looks like everything is still working from here. CI is also green, but requires approval to run a few more checks.

File: aws/resource_aws_ebs_snapshot_import_test.go
Comment (DrFaust92) [inline aws/resource_aws_ebs_snapshot_import_test.go:42]: Add tag update scenario here based on https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/contribution-checklists.md#resource-tagging-acceptance-testing-implementation

File: aws/resource_aws_ebs_snapshot_import.go
Comment (DrFaust92) [inline aws/resource_aws_ebs_snapshot_import.go:154]: support default tags:
https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/contribution-checklists.md#resource-tagging-code-implementation

**Review by DrFaust92 — CHANGES_REQUESTED** [review]

**Comment by cole-h** [comment]
Once again, thanks for your time and review, @DrFaust92! I've fixed your suggestions in a few new commits, and `testacc` still passes.

**Comment by cole-h** [comment]
Rebased to resolve a conflict.

**Comment by grahamc** [comment]
:heart:

**Comment by grahamc** [comment]
:partying_face: wow, thank you so much Bill!

**Pull request merged**
